src/lib/import-export/import/handlers/backup-handler-factory.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import { BackupArchiveInfo } from '../types';
+import { BackupHandlerSql } from './backup-handler-sql';
+import { BackupHandlerTarGz } from './backup-handler-tar-gz';
+import { BackupHandlerZip } from './backup-handler-zip';
+
+export interface BackupHandler {
+	listFiles( file: BackupArchiveInfo ): Promise< string[] >;
+	extractFiles( file: BackupArchiveInfo, extractionDirectory: string ): Promise< void >;
+}
+
+export class BackupHandlerFactory {
+	private static zipTypes = [
+		'application/zip',
+		'application/x-zip',
+		'application/x-zip-compressed',
+		'application/octet-stream',
+	];
+	private static zipExtensions = [ '.zip' ];
+
+	private static tarGzTypes = [
+		'application/gzip',
+		'application/x-gzip',
+		'application/x-gtar',
+		'application/x-tgz',
+		'application/x-compressed-tar',
+		'application/tar+gzip',
+	];
+	private static tarGzExtensions = [ '.tar.gz', '.tgz' ];
+
+	private static sqlTypes = [
+		'application/sql',
+		'application/x-sql',
+		'text/sql',
+		'text/x-sql',
+		'text/plain',
+	];
+	private static sqlExtensions = [ '.sql' ];
+
+	static create( file: BackupArchiveInfo ): BackupHandler {
+		if ( this.isZip( file ) ) {
+			return new BackupHandlerZip();
+		} else if ( this.isTarGz( file ) ) {
+			return new BackupHandlerTarGz();
+		} else if ( this.isSql( file ) ) {
+			return new BackupHandlerSql();
+		} else {
+			throw new Error( 'Unsupported file format. Only zip, tar.gz, and sql files are supported.' );
+		}
+	}
+
+	private static isZip( file: BackupArchiveInfo ): boolean {
+		return (
+			this.zipTypes.includes( file.type ) &&
+			this.zipExtensions.some( ( ext ) => file.path.endsWith( ext ) )
```

Comment:
I wonder if it's redundant to check the file extension if we are already checking the MIME type of the file. WDYT?

Reply:
It is a bit,but originally I thought that many users are not familiar with MIME types, but rather with file extensions. Checking ext as well will make more clear for those users what we want.
What do you think?

Reply:
Definitely, we should filter by extension in the file dialog when importing a file, but I presume that it will be handled in a different place. If I understand this part correctly, these conditions are mainly used to determine the handler to import the file, which might be agnostic to extension. In any case, I don't have a strong preference to remove the extension condition so we can keep it. My comment is mainly driven by the case of dropping a file. Probably this is an edge case, but if the user drops a backup file without extension, should we process it if it has a valid MIME type or discard it?

Reply:
Ah, good point. In that case, let's keep the condition that checks the file extension 👍 .